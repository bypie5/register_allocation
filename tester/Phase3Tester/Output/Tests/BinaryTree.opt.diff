--- SelfTestCases/BinaryTree.opt.out	2011-04-28 14:55:20.000000000 -0700
+++ ./Output/Tests/BinaryTree.opt.out	2020-02-24 15:38:10.000000000 -0800
@@ -1,26 +1 @@
-16
-100000000
-8
-16
-4
-8
-12
-14
-16
-20
-24
-28
-1
-1
-1
-0
-1
-4
-8
-14
-16
-20
-24
-28
-0
-0
+Program has no "Main" function.
