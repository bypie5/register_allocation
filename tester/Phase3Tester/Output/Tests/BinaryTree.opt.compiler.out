func Main [in 0, out 0, local 0]
    $s0 = 
    PrintIntS($s0)
    ret
func BT.Start [in 0, out 0, local 0]
    $s0 = HeapAllocZ(24)
    Error()
    $s1 = 
    Error()
    $s1 = 
    PrintIntS(100000000)
    Error()
    $s1 = 
    Error()
    $s1 = 
    Error()
    $s1 = 
    Error()
    $s1 = 
    Error()
    $s1 = 
    Error()
    $s1 = 
    Error()
    $s1 = 
    Error()
    $s1 = 
    Error()
    $s1 = 
    Error()
    $s1 = 
    PrintIntS($s1)
    Error()
    $s1 = 
    PrintIntS($s1)
    Error()
    $s1 = 
    PrintIntS($s1)
    Error()
    $s0 = 
    PrintIntS($s0)
    Error()
    $s0 = 
    PrintIntS($s0)
    Error()
    $s0 = 
    Error()
    $s0 = 
    Error()
    $s0 = 
    PrintIntS($s0)
    ret
func Tree.Init [in 0, out 0, local 0]
    ret
func Tree.SetRight [in 0, out 0, local 0]
    ret
func Tree.SetLeft [in 0, out 0, local 0]
    ret
func Tree.GetRight [in 0, out 0, local 0]
    ret
func Tree.GetLeft [in 0, out 0, local 0]
    ret
func Tree.GetKey [in 0, out 0, local 0]
    ret
func Tree.SetKey [in 0, out 0, local 0]
    ret
func Tree.GetHas_Right [in 0, out 0, local 0]
    ret
func Tree.GetHas_Left [in 0, out 0, local 0]
    ret
func Tree.SetHas_Left [in 0, out 0, local 0]
    ret
func Tree.SetHas_Right [in 0, out 0, local 0]
    ret
func Tree.Compare [in 0, out 0, local 0]
    $s0 = 0
    $s0 = Add($s1 1)
    $s3 = LtS($s2 $s1)
    $s0 = 0
    $s2 = LtS($s1 $s0)
    $s0 = 0
    $s0 = 1
    ret
func Tree.Insert [in 0, out 0, local 0]
    $s1 = HeapAllocZ(24)
    Error()
    $s3 = 
    $s0 = this
    $s3 = 1
    Error()
    $s4 = 
    $s5 = LtS($s2 $s4)
    Error()
    $s4 = 
    Error()
    $s0 = 
    $s3 = 0
    Error()
    $s4 = 
    Error()
    $s4 = 
    Error()
    $s4 = 
    Error()
     = 
     = 0
    Error()
    $s0 = 
    Error()
    $s0 = 
    ret
func Tree.Delete [in 0, out 0, local 0]
    $s0 = $s2
    $s3 = $s2
    $s4 = 1
    $s5 = 0
    $s6 = 1
    Error()
    $s7 = 
    $s6 = LtS($s1 $s7)
    Error()
    $s6 = 
    $s3 = $s0
    Error()
    $s0 = 
    $s4 = 0
    $t0 = LtS($s6 $s1)
    Error()
    $s6 = 
    $s3 = $s0
    Error()
    $s0 = 
     = 0
    Error()
    $s4 = 
    Error()
    $s0 = 
    $s0 = 1
    $s0 = 
    $s0 = 
    $s0 = 1
    $s1 = 0
    $s1 = 0
    ret
func Tree.Remove [in 0, out 0, local 0]
    Error()
    $s3 = 
    $s0 = 
    Error()
    $s2 = 
    $s0 = 
    Error()
    $s2 = 
    Error()
    $s3 = 
    Error()
    $s2 = 
    $s3 = 
    Error()
    $s1 = 
    Error()
    $s0 = 
    Error()
    $s2 = 
    Error()
    $s0 = 
    ret
func Tree.RemoveRight [in 0, out 0, local 0]
    Error()
    $s3 = 
    Error()
    Error()
    $s0 = 
    Error()
    $s0 = 
    $s3 = 
    $s0 = $s1
    Error()
     = 
    Error()
    $s2 = 
    Error()
    $s0 = 
    ret
func Tree.RemoveLeft [in 0, out 0, local 0]
    Error()
    $s3 = 
    Error()
    Error()
    $s0 = 
    Error()
    $s0 = 
    $s3 = 
    $s0 = $s1
    Error()
     = 
    Error()
    $s2 = 
    Error()
    $s0 = 
    ret
func Tree.Search [in 0, out 0, local 0]
    $s1 = $s0
    $s0 = 1
    $s3 = 0
    Error()
    $s4 = 
    $s5 = LtS($s2 $s4)
    Error()
    $s4 = 
    Error()
    $s1 = 
    $s0 = 0
    $s5 = LtS($s4 $s2)
    Error()
    $s0 = 
    Error()
     = 
     = 0
     = 1
    $s0 = 0
    ret
func Tree.Print [in 0, out 0, local 0]
    $s0 = $s1
    $s2 = 
    ret
func Tree.RecPrint [in 0, out 0, local 0]
    Error()
    $s2 = 
    Error()
    $s2 = 
    $s3 = 
     = 1
    Error()
    $s2 = 
    PrintIntS($s2)
    Error()
    $s0 = 
    Error()
    $s0 = 
    $s1 = 
    $s0 = 1
    ret

